pkg/stacks/thanos/register_candidate.go

```diff
@@ -21,6 +21,38 @@ import (
 	"github.com/tokamak-network/trh-sdk/pkg/utils"
 )
 
+func (t *ThanosStack) checkAdminBalance(ctx context.Context, adminAddress ethCommon.Address, amount float64, l1Client *ethclient.Client) error {
+	fmt.Printf("Checking admin's TON token balance... \n")
+
+	chainIDFromClient, errChainID := l1Client.ChainID(ctx)
+	if errChainID != nil {
+		return fmt.Errorf("failed to get L1 chain ID: %w", errChainID)
+	}
+
+	chainConfig := constants.L1ChainConfigurations[chainIDFromClient.Uint64()]
+
+	tonAddress := ethCommon.HexToAddress(chainConfig.TON)
+	tokenInstance, errToken := abis.NewTON(tonAddress, l1Client)
+	if errToken != nil {
+		return fmt.Errorf("failed to instantiate TON token contract at %s: %w", tonAddress.Hex(), errToken)
+	}
+
+	adminBalance, errBalance := tokenInstance.BalanceOf(&bind.CallOpts{Context: ctx}, adminAddress)
+	if errBalance != nil {
+		return fmt.Errorf("failed to get TON balance for admin %s from contract %s: %w", adminAddress, tonAddress.Hex(), errBalance)
+	}
+	amountInWei := new(big.Float).Mul(big.NewFloat(amount), big.NewFloat(1e18))
+	requiredAmountBigInt, _ := amountInWei.Int(nil)
+
+	if adminBalance.Cmp(requiredAmountBigInt) < 0 {
```

Comment:
Suggestion for UX: How about give an error message after getting the amount of TON to stake if admin's TON balance is insufficient?